lib/HLSL/DxilValidation.cpp

```diff
@@ -1131,6 +1131,112 @@ static int GetCBufSize(Value *cbHandle, ValidationContext &ValCtx) {
+    const Type *pNodeHandleTy = ValCtx.DxilMod.GetOP()->GetNodeHandleType();
+    const Type *pNodeRecordHandleTy =
+        ValCtx.DxilMod.GetOP()->GetNodeRecordHandleType();
+
+    const Type *argTy = op->getType();
+    if (argTy == pNodeHandleTy || argTy == pNodeRecordHandleTy ||
+        argTy == pHandleTy) {
+      bool bIsUninitialized =
+          isa<UndefValue>(op) || isa<ConstantAggregateZero>(op);
+      if (argTy == pHandleTy) {
+        hlsl::DxilResourceProperties DRP = GetResourceFromHandle(op, ValCtx);
+        if (!DRP.isValid()) {
+          ValCtx.EmitInstrError(CI, ValidationRule::InstrResourceInvalid);
+        }
+      }
+      if (bIsUninitialized) {
+        ValCtx.EmitInstrError(CI, ValidationRule::InstrNoReadingUninitialized);
+      }
+    }
+  }
+}
+
+void ValidateHandleArgs(CallInst* CI, DXIL::OpCode opcode, ValidationContext &ValCtx) {
+
+  switch (opcode) {
+  // handle ops
+  case DXIL::OpCode::CreateHandle:
+  case DXIL::OpCode::CBufferLoad:
+  case DXIL::OpCode::CBufferLoadLegacy:
+  case DXIL::OpCode::Sample:
+  case DXIL::OpCode::SampleBias:
+  case DXIL::OpCode::SampleLevel:
+  case DXIL::OpCode::SampleGrad:
+  case DXIL::OpCode::SampleCmp:
+  case DXIL::OpCode::SampleCmpLevelZero:
+  case DXIL::OpCode::TextureLoad:
+  case DXIL::OpCode::TextureStore:
+  case DXIL::OpCode::BufferLoad:
+  case DXIL::OpCode::BufferStore:
+  case DXIL::OpCode::BufferUpdateCounter:
+  case DXIL::OpCode::GetDimensions:
+  case DXIL::OpCode::TextureGather:
+  case DXIL::OpCode::TextureGatherCmp:
+  case DXIL::OpCode::Texture2DMSGetSamplePosition:
+  case DXIL::OpCode::AtomicBinOp:
+  case DXIL::OpCode::AtomicCompareExchange:
+  case DXIL::OpCode::CalculateLOD:
+  case DXIL::OpCode::RawBufferLoad:
+  case DXIL::OpCode::RawBufferStore:
+  case DXIL::OpCode::TraceRay:
+  case DXIL::OpCode::WriteSamplerFeedback:
+  case DXIL::OpCode::WriteSamplerFeedbackBias:
+  case DXIL::OpCode::WriteSamplerFeedbackLevel:
+  case DXIL::OpCode::WriteSamplerFeedbackGrad:
+  case DXIL::OpCode::RayQuery_TraceRayInline:
+  case DXIL::OpCode::CreateHandleFromBinding:
+  case DXIL::OpCode::CreateHandleFromHeap:
+  case DXIL::OpCode::TextureGatherRaw:
+  case DXIL::OpCode::SampleCmpLevel:
+  case DXIL::OpCode::TextureStoreSample:
+  case DXIL::OpCode::WaveMatrix_LoadRawBuf:
+  case DXIL::OpCode::WaveMatrix_StoreRawBuf:
+  case DXIL::OpCode::BarrierByMemoryHandle:
+  // node handle ops
+  case DXIL::OpCode::IncrementOutputCount:
+  case DXIL::OpCode::CreateNodeOutputHandle:
+  case DXIL::OpCode::IndexNodeHandle:
+  case DXIL::OpCode::NodeOutputIsValid:
+  // node record handle ops
+  case DXIL::OpCode::GetNodeRecordPtr:
+  case DXIL::OpCode::OutputComplete:
+  case DXIL::OpCode::GetInputRecordCount:
+  case DXIL::OpCode::FinishedCrossGroupSharing:
+  case DXIL::OpCode::BarrierByNodeRecordHandle:
+  case DXIL::OpCode::CreateNodeInputRecordHandle:
+  // TODO: add case DXIL::OpCode::IndexNodeRecordHandle:
```

Comment:
I see that this and other TODOs were to be addressed in a followup PR. Could you create an issue and reference it here to make sure that isn't lost and all the information is stored?